pennylane/beta/plugins/default_tensor.py

```diff
@@ -428,38 +422,167 @@ def ev(self, obs_nodes, wires, contraction_method="auto"):
+                tn.connect(bra[w], obs_node[output_idx])  # <psi|A
+        # unmeasured wires are contracted directly between bra and ket
+        for w in set(all_wires) - set(meas_wires):
+            tn.connect(bra[w], ket[w])
+
+        # At this stage, all nodes are connected, and the contraction yields a
+        # scalar value.
+        ket_and_observable_node = ket
+        for obs_node in obs_nodes:
+            ket_and_observable_node = tn.contract_between(obs_node, ket_and_observable_node)
+        return tn.contract_between(bra, ket_and_observable_node).tensor
+
+    def _ev_mps(self, obs_nodes, wires):
+        r"""Expectation value of observables on specified wires using a MPS representation.
 
-        Args:
-            contraction_method (str): The contraction method to be employed.
-                Possible choices are "auto", "greedy", "branch", or "optimal".
-                See TensorNetwork library documentation for more details.
+         Args:
+            obs_nodes (Sequence[tn.Node]): the observables as TensorNetwork Nodes
+            wires (Sequence[Sequence[int]]): measured subsystems for each observable
+         Returns:
+            complex: expectation value :math:`\expect{A} = \bra{\psi}A\ket{\psi}`
+        """
+        if any(len(wires_seq) > 2 for wires_seq in wires):
+            raise NotImplementedError(
+                "Multi-wire measurement only supported for nearest-neighbour wire pairs."
+            )
+        else:
+            if len(obs_nodes) == 1 and len(wires[0]) == 1:
+                # TODO: can measure multiple local expectation values at once,
+                # but this would require change of `expval` behaviour and
+                # refactor of `execute` logic from parent class
+                expval = self.mps.measure_local_operator(obs_nodes, wires[0])[0]
+            else:
+                conj_nodes = [tn.conj(node) for node in self.mps.nodes]
+                meas_wires = []
+                # connect measured bra and ket nodes with observables
+                for obs_node, wire_seq in zip(obs_nodes, wires):
+                    if len(wire_seq) == 2 and abs(wire_seq[0] - wire_seq[1]) > 1:
+                        raise NotImplementedError(
+                            "Multi-wire measurement only supported for nearest-neighbour wire pairs."
+                        )
+                    offset = len(wire_seq)
+                    for idx, wire in enumerate(wire_seq):
+                        tn.connect(conj_nodes[wire][1], obs_node[idx])
+                        tn.connect(obs_node[offset + idx], self.mps.nodes[wire][1])
+                    meas_wires.extend(wire_seq)
+                for wire in range(self.num_wires):
+                    # connect unmeasured ket nodes with bra nodes
+                    if wire not in meas_wires:
+                        tn.connect(conj_nodes[wire][1], self.mps.nodes[wire][1])
```

Comment:
From my understanding, contraction is not needed between unmeasured modes as long as the MPS state is canonicalized? Indices where the observable is not measured are simply ignored, and only measured indices are contracted with the Hermitian observable.

Reply:
Not too sure, I'll have to look into this